### PR TITLE
Add support for multiple schemas in Metadata Editor

### PR DESCRIPTION
**Pull request by ajbozarth**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ptitzler** [comment]
For the runtime configuration type specific display in

image

where you are going to replace  "Kubeflow Pipelines UI" with the appropriate text for the runtime type, would it also be possible to display additional information based on the type?

For Airflow, a user would want to have access to the configured github repository where the DAGs are stored. The URL can be assembled from the information provided in the runtime configuration. 

image


Is something like this already possible purely based on the existing metadata or is the metadata editor too limited?

**Comment by ajbozarth** [comment]
I am going to update that text based on the schema. As for having different info depending on the schema, that would be a "new feature" since the displayed info is currently hard-coded, but it would be easy to add a uihint value that could determine which properties are shown instead of hard-coding it. I will look into that when I get back to that part of the code

**Comment by ajbozarth** [comment]
Some To Do's:
- Add drop down to `+` button
- Remove hard-coded schema references
- update UI to display correct content in sidebar

**Comment by ptitzler** [comment]
User scenarios we need to verify:

- [ ] User can choose which type of runtime configuration to create
- [ ] User can create/edit/delete a KFP runtime configuration
- [ ] User can create/edit/delete an Apache Airflow runtime configuration

**Comment by ajbozarth** [comment]
Still To Do:
- [ ] Replace double `+` icons with a drop down
- [ ] Fix command pallets (could be done in followup PR) #1298
- [ ] Customize Runtimes UI content based on schema (prob in a followup PR) https://github.com/elyra-ai/elyra/pull/1327#issuecomment-785325718
- [ ] Update tests

**Comment by ptitzler** [comment]
Turns out I forgot to list a user workflow. The "Open Runtimes" button in the Pipeline Editor doesn't work and needs to be added to the TODO list.

Tested:
- User can choose which type of runtime configuration to create
- User can create/edit/delete a KFP runtime configuration
- User can create/edit/delete an Apache Airflow runtime configuration

With the (for now) unified API Endpoint label it's impossible to determine the runtime type without opening the configuration if the user chooses a generic name and description:
  
image

Is there a way to expose the type information to solve the dilemma?

**Comment by ajbozarth** [comment]
I pushed an update addressing Patricks previous comments. 

I've also decided on using Material-UI's ButtonGroup library to handle the dropdown and will be implementing that tomorrow

**Comment by ptitzler** [comment]
Thanks Alex! I was now able to complete all scenarios.

**Comment by ptitzler** [comment]
We should probably revisit the label text in the future.

image

**Comment by ajbozarth** [comment]
I've pushed a working implementation of the drop down, there's still a few things to clean up, specifically it opens a drop down when there's only one schema (it should just open the new editor instead), the text for the tooltips and dropdown are a bit verbose, and lastly I want to clean up and streamline the new code itself. Also I still need to deal with the tests

**Comment by bourdakos1** [comment]
actually, that might be easier said then done... it looks like this widget is a lot more complicated than I thought at first glance haha

**Comment by bourdakos1** [comment]
I think the bigger issue here is that we do a lot of inheritance, when react places nicer with composition

File: packages/metadata-common/src/MetadataWidget.tsx [hunk 1]
```diff
@@ -308,14 +319,83 @@ export class MetadataDisplay<
   }
 }
 
+interface IButtonProps {
+  schemas: IDictionary<any>[];
+  addMetadata: (schema: string) => void;
+}
+
+// eslint-disable-next-line func-style
+function AddMetadataButton(props: IButtonProps): React.ReactElement {
```
Comment (lresende) [inline packages/metadata-common/src/MetadataWidget.tsx:328]: Could we use the arrow function style?

**Comment by ptitzler** [comment]
If no runtime schema is defined, the following error is raised:

image

Note that this is an artificial scenario. I removed the schema files in `elyra/metadata/schemas/` to simulate the use case where a customer wants to prevent users from creating/using any runtime configuration.

**Comment by ptitzler** [comment]
OK to defer to later:
 - manually add a third runtime configuration schema
 - build elyra
 - create a runtime configuration using the new type
 - in the list of runtime configurations the new runtime configuration is displayed as a kubeflow  configuration

File: packages/metadata-common/src/MetadataWidget.tsx (hunk 1 above)
Comment (ajbozarth) [reply packages/metadata-common/src/MetadataWidget.tsx]: done

**Comment by ptitzler** [comment]
> If no runtime schema is defined, the following error is raised:
...

Opened https://github.com/elyra-ai/elyra/issues/1360.

**Comment by karlaspuldaro** [comment]
Just built the latest changes and I see some react warnings in the chrome dev console related to the `AddMetadataButton` component
image

**Comment by ajbozarth** [comment]
I just pushed an update that fixes current tests (does not add new ones though)

**Review by ptitzler — APPROVED** [review]
LGTM. Excluding issues that were deferred.

**Comment by ajbozarth** [comment]
I'll look into the chrome errors, I dont believe I was seeing them in safari.

Also I still need to get the dropdown to support dark mode

**Review by lresende — APPROVED** [review]

**Pull request merged**